core/controllers/profile.py

```diff
@@ -361,6 +364,7 @@ def post(self):
 class SiteLanguageHandler(base.BaseHandler):
     """Changes the preferred system language in the user's preferences."""
 
+    @acl_decorators.open_access
```

Comment:
Yep. I did that. But, there is a test (test_save_site_language_no_user in profile_test.py) that requires that non logged in user should be able to send a request to this handler. So, I reverted it.